cmake/D2D_Compilers.cmake

```diff
@@ -97,3 +97,16 @@ option(SINGLE_PRECISION_OUTPUT "Build XCompact with output in single precision"
 if (SINGLE_PRECISION_OUTPUT)
   add_definitions("-DSAVE_SINGLE")
```

Comment:
This should be moved and changed to `target_compile_definitions(decomp2d PUBLIC -DSAVE_SINGLE)` ?

Reply:
Some preprocessor variables are used in the examples. We can move some preprocessor variables to `src/CMakeLists.txt` but not all of them. If we move all of them, we should add the variables in the CMakeLists.txt of the example

Reply:
Got the following error when compiling 
```
Building Fortran object src/CMakeFiles/decomp2d.dir/io.f90.o
cd /Users/rfj82982/GIT_projects/2decomp-fft_cedric/build/src && /opt/local/bin/mpif90-mpich-gcc12 -DADIOS2 -DADIOS2_USE_MPI -DDOUBLE_PREC -DVERSION=\"e85638f\" -I/Users/rfj82982/GIT_projects/2decomp-fft_cedric/src -I/Users/rfj82982/GIT_projects/2decomp-fft_cedric/build/include -I/opt/local/include/mpich-mp -I/Users/rfj82982/GIT_projects/ADIOS2/build/opt/include/adios2/fortran -cpp -std=f2008 -ffree-line-length-none -fallow-argument-mismatch -O3 -march=native -J../include -c /Users/rfj82982/GIT_projects/2decomp-fft_cedric/src/io.f90 -o CMakeFiles/decomp2d.dir/io.f90.o
/Users/rfj82982/GIT_projects/2decomp-fft_cedric/src/io.f90:140:91:

  140 |       call adios2_init(adios, trim(config_file), decomp_2d_comm, adios2_debug_mode, ierror)
      |                                                                                           1
Error: There is no specific subroutine for the generic 'adios2_init' at (1)
make[2]: *** [src/CMakeFiles/decomp2d.dir/io.f90.o] Error 1
make[1]: *** [src/CMakeFiles/decomp2d.dir/all] Error 2
make: *** [all] Error 2
```
I’ll check if i can fix it.

Reply:
I need to check this ADIOS 2.9. I am on this repo 
https://github.com/ornladios/ADIOS2.git
on the main branch. Just did a pull to have the most recent version and build/install. 

I cannot find adios2_init among the stuffs in the 
include/adios2/fortran

Reply:
I moved from the master to the 2.9.0 branch and I have also tested the tar file. In all cases I always get the error above.

Reply:
It looks like that the arguments for `adios2_init` are now different from what we have in our call 
https://adios2.readthedocs.io/en/latest/api_full/api_full.html#fortran-bindings
```
! MPI versions
subroutine adios2_init(adios, comm, ierr)
subroutine adios2_init(adios, config_file, comm, ierr)

! Non-MPI serial versions
subroutine adios2_init(adios, ierr)
subroutine adios2_init(adios, config_file, ierr)
```
while we have
`call adios2_init(adios, trim(config_file), decomp_2d_comm, adios2_debug_mode, ierror)`

Reply:
I have tested the present PR against ADIOS version 2.8.

Reply:
https://github.com/ornladios/ADIOS2/pull/3204

Reply:
Fixed in https://github.com/2decomp-fft/2decomp-fft/pull/195/commits/3394db2d617b68d7e9a7f9f82a9bbd4460fcb630